### PR TITLE
feat: add formulas 8.81 to 8.85 from FprEN 1992-1-1:2023 clause 8.3.4

### PR DESCRIPTION
**Pull request by egarciamendez**
Adds all five numbered formulas of FprEN 1992-1-1:2023 clause 8.3.4, "Torsional resistance of compact or closed sections", paragraphs (1) to (4). The clause is complete with this pull request.

(8.81) is the design torsional capacity, the minimum of the three resistances of (8.82), (8.83) and (8.84): yielding of the shear reinforcement, yielding of the longitudinal reinforcement, and crushing of the compression field in concrete. (8.85) is the range the standard gives for the inclination of the compression field used in those three. All of it applies to a single cell, thin-walled section or a sub-section with constant effective wall thickness.

Decisions a reviewer has to weigh:

- **(8.85) uses `AggregatedComparisonFormula`, not `DoubleComparisonFormula`.** The repository holds both for this shape: (8.41) uses the latter, (8.58) the former. The deciding difference is that `DoubleComparisonFormula` does not inherit from `ComparisonFormula` and therefore has no `unity_check` at all, so `Form8Dot41InclinationCompressionField.unity_check` raises `AttributeError`. Modelled as two aggregated comparisons, each bound of (8.85) keeps its own unity check and the aggregate is the governing one. (8.41) is left alone.
- **(8.85) is not a duplicate of (8.41)**, although both are bounded above by `cot θ_min` of 8.2.3(4). The lower bound here is `1/cot θ_min` instead of the constant 1, so torsion permits a flatter compression field than shear does, symmetric in `cot θ` about 1. With `cot θ_min = 2.5` the range on `cot θ` is `[0.4, 2.5]`, and both unity checks read 0.4 at `cot θ = 1`.
- If `cot θ_min < 1` the printed lower bound exceeds the printed upper bound and nothing satisfies the relation. The class reports that rather than reordering the two sides, and a test pins it. 8.2.3(4) puts `cot θ_min` at 2,5 or higher, so it does not arise in practice.
- The standard prints the numerator of (8.83) as the single quantity `ΣA_sl f_yd`, the yield force of the longitudinal reinforcement. This class takes the area and the stress separately, which mirrors (8.82) and lets `complete_with_units` expose a unit mismatch. The two are equal only when every bar counted carries the same `f_yd`, and the class docstring says so.
- All angles are taken in degrees rather than as a pre-computed cotangent, following the choice made for (8.55) to (8.57).
- Formula (8.85) is a recommendation for usual cases, and enforcing it inside (8.82) to (8.84) would need `cot θ_min`, which none of those classes has. It is documented on the `theta` parameter of each and left as a separate check.
- `nu` in (8.84) is left entirely to the caller. The standard points at Annex G and offers 0,60 for `cot θ = 1,0`; neither is enforced.
- (8.84) is tested at two angles. Its denominator `cot θ + tan θ` is symmetric about 45 degrees, so a single angle cannot distinguish the sum of the two functions from a multiple of one of them.
- (8.81) keeps the printed `min{a; b; c}` with braces and semicolons instead of the `\min\left(a, b\right)` used elsewhere in the repository. Those other uses are all clamps on a computed value, not a min-set the standard prints.
- The docs table still shows (8.79) and (8.80) as not implemented, because this branch starts from main and those two sit in #1135.

## Formulas as implemented

**Formula (8.81)**, `Form8Dot81DesignTorsionalCapacity`

`equation`

$$
\tau_{t,Rd} = \min\left\lbrace \tau_{t,Rd,sw}; \tau_{t,Rd,sl}; \tau_{t,Rd,max} \right\rbrace
$$

`complete`

$$
\tau_{t,Rd} = \min\left\lbrace \tau_{t,Rd,sw}; \tau_{t,Rd,sl}; \tau_{t,Rd,max} \right\rbrace = \min\left\lbrace 3.625; 2.900; 4.000 \right\rbrace = 2.900 \ MPa
$$

`complete_with_units`

$$
\tau_{t,Rd} = \min\left\lbrace \tau_{t,Rd,sw}; \tau_{t,Rd,sl}; \tau_{t,Rd,max} \right\rbrace = \min\left\lbrace 3.625 \ MPa; 2.900 \ MPa; 4.000 \ MPa \right\rbrace = 2.900 \ MPa
$$

**Formula (8.82)**, `Form8Dot82TorsionalStressResistanceShearReinforcement`

`equation`

$$
\tau_{t,Rd,sw} = \cot(\theta) \cdot \frac{A_{sw}}{t_{eff} \cdot s} \cdot f_{ywd}
$$

`complete`

$$
\tau_{t,Rd,sw} = \cot(\theta) \cdot \frac{A_{sw}}{t_{eff} \cdot s} \cdot f_{ywd} = \cot(26.565) \cdot \frac{100.000}{120.000 \cdot 200.000} \cdot 435.000 = 3.625 \ MPa
$$

`complete_with_units`

$$
\tau_{t,Rd,sw} = \cot(\theta) \cdot \frac{A_{sw}}{t_{eff} \cdot s} \cdot f_{ywd} = \cot(26.565 ^\circ) \cdot \frac{100.000 \ mm^2}{120.000 \ mm \cdot 200.000 \ mm} \cdot 435.000 \ MPa = 3.625 \ MPa
$$

**Formula (8.83)**, `Form8Dot83TorsionalStressResistanceLongitudinalReinforcement`

`equation`

$$
\tau_{t,Rd,sl} = \frac{\Sigma A_{sl} \cdot f_{yd}}{t_{eff} \cdot u_k \cdot \cot(\theta)}
$$

`complete`

$$
\tau_{t,Rd,sl} = \frac{\Sigma A_{sl} \cdot f_{yd}}{t_{eff} \cdot u_k \cdot \cot(\theta)} = \frac{2400.000 \cdot 435.000}{120.000 \cdot 1500.000 \cdot \cot(26.565)} = 2.900 \ MPa
$$

`complete_with_units`

$$
\tau_{t,Rd,sl} = \frac{\Sigma A_{sl} \cdot f_{yd}}{t_{eff} \cdot u_k \cdot \cot(\theta)} = \frac{2400.000 \ mm^2 \cdot 435.000 \ MPa}{120.000 \ mm \cdot 1500.000 \ mm \cdot \cot(26.565 ^\circ)} = 2.900 \ MPa
$$

**Formula (8.84)**, `Form8Dot84TorsionalStressResistanceConcreteCrushing`

`equation`

$$
\tau_{t,Rd,max} = \frac{\nu \cdot f_{cd}}{\cot(\theta) + \tan(\theta)}
$$

`complete`

$$
\tau_{t,Rd,max} = \frac{\nu \cdot f_{cd}}{\cot(\theta) + \tan(\theta)} = \frac{0.500 \cdot 20.000}{\cot(26.565) + \tan(26.565)} = 4.000 \ MPa
$$

`complete_with_units`

$$
\tau_{t,Rd,max} = \frac{\nu \cdot f_{cd}}{\cot(\theta) + \tan(\theta)} = \frac{0.500 \cdot 20.000 \ MPa}{\cot(26.565 ^\circ) + \tan(26.565 ^\circ)} = 4.000 \ MPa
$$

**Formula (8.85)**, `Form8Dot85CheckCotangentCompressionFieldTorsion`

`equation`

$$
CHECK \to \frac{1}{\cot(\theta_{min})} \leq \cot(\theta) \leq \cot(\theta_{min})
$$

`complete`

$$
CHECK \to \frac{1}{\cot(\theta_{min})} \leq \cot(\theta) \leq \cot(\theta_{min}) \to \frac{1}{\cot(21.801)} \leq \cot(26.565) \leq \cot(21.801) \to OK
$$

`complete_with_units`

$$
CHECK \to \frac{1}{\cot(\theta_{min})} \leq \cot(\theta) \leq \cot(\theta_{min}) \to \frac{1}{\cot(21.801 ^\circ)} \leq \cot(26.565 ^\circ) \leq \cot(21.801 ^\circ) \to OK
$$

Contributes to #1083

## Summary by CodeRabbit

* **New Features**
  * Added Eurocode torsional design calculations for formulas 8.81–8.85.
  * Added checks for torsional capacity, reinforcement resistance, concrete crushing, and compression-field inclination.
  * Added symbolic, numeric, and unit-aware LaTeX representations for the calculations.

* **Documentation**
  * Marked formulas 8.81–8.85 as complete and linked them to their implementations.

* **Tests**
  * Added coverage for calculations, input validation, boundary conditions, angle handling, and formatted output.

### CODE REVIEW

File: blueprints/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_85.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+        r"""Check the selected inclination of the compression field against its lower bound.
+
+        FprEN 1992-1-1:2023 (E) art. 8.3.4(4) - Formula (8.85)
+
+        Parameters
+        ----------
+        theta : DEG
+            [$\theta$] Selected inclination of the compression field [$degrees$].
+        theta_min : DEG
+            [$\theta_{min}$] Minimal inclination of the compression field according to 8.2.3(4) [$degrees$].
```
Comment (casper-van-der-krift) [inline blueprints/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_85.py:31]: In section 8.2.3(4) theta_min is referenced by formula 8.41. The text under this formula provides information on how to choose theta_min. In our formula 8.41 these guidelines are present in the docstring of the constructor. Since we are using the same theta_min, I would suggest to also incorporate these guidelines here (copy from formula 8.41 is sufficient).

Image

File: blueprints/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_85.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+            return_symbol=r"CHECK",
+            result="OK" if self.__bool__() else r"\text{Not OK}",
+            equation=_equation,
+            numeric_equation=latex_replace_symbols(template=_equation, replacements=_replacements, unique_symbol_check=False),
+            comparison_operator_label=r"\to",
+            unit="",
+        )
+
+
+class Form8Dot85CheckCotangentCompressionFieldTorsion(AggregatedComparisonFormula):
```
Comment (casper-van-der-krift) [inline blueprints/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_85.py:123]: An object instance of this class is of type float, while I would excpect a boolean to indicate whether the provided value of theta is within the given bounds.

Image

**Review by casper-van-der-krift — CHANGES_REQUESTED** [review]

File: tests/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/test_formula_8_85.py [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+            (THETA_COT_0_25, False),  # flatter than the lower bound allows
+        ],
+    )
+    def test_evaluation(self, theta: float, expected: bool) -> None:
+        """Tests the evaluation of the result."""
+        # Create object to test
+        formula = Form8Dot85CheckCotangentCompressionFieldTorsion(theta=theta, theta_min=THETA_COT_2_5)
+
+        # Perform test by assert
+        assert bool(formula) is expected
```
Comment (casper-van-der-krift) [inline tests/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/test_formula_8_85.py:42]: I don't think we should have to call bool(formula) to get a boolean that indicates whether theta is within range. Perhaps use __repr__ to obtain this implicitly.

**Review by casper-van-der-krift — CHANGES_REQUESTED** [review]

File: blueprints/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_85.py (hunk 2 above)
Comment (egarciamendez) [reply blueprints/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_85.py:123]: @SZeltaat to close this PR we need to solve issue #1160 first. 
Shall I do that? will you do that?

Comment (SZeltaat) [reply blueprints/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_85.py:123]: I'm working on it 😉

Comment (SZeltaat) [reply blueprints/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_85.py:123]: Done! See #1168. Please feel free to make changes and/or merge the PR.

File: blueprints/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_85.py [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+
+    @staticmethod
+    def _bounds(theta: DEG, theta_min: DEG) -> Sequence[ComparisonFormula]:
+        """Builds the two halves of the printed range."""
+        return (
+            SubForm8Dot85LowerBound(theta=theta, theta_min=theta_min),
+            SubForm8Dot85UpperBound(theta=theta, theta_min=theta_min),
+        )
+
+    def __new__(cls, theta: DEG, theta_min: DEG) -> Self:
```
Comment (SZeltaat) [inline blueprints/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_85.py:170]: As per #1168 please override _define_aggregation instead of dunder new. This breaks part of the functionality of AggregatedComparisonFormula.

File: blueprints/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_85.py (hunk 1 above)
Comment (egarciamendez) [reply blueprints/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_85.py:30]: Added, copied the theta_min guidance from 8.41's docstring into all three classes.

File: blueprints/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_85.py (hunk 4 above)
Comment (egarciamendez) [reply blueprints/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_85.py]: Done, now overrides `_define_aggregation` instead of `__new__`, matching #1168.

File: blueprints/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_85.py (hunk 2 above)
Comment (egarciamendez) [reply blueprints/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_85.py]: This is the base ComparisonFormula/AggregatedComparisonFormula design (float-based), used identically in 8.41, 8.58, 8.60, 8.69 and 8.70. Changing it would be a base-class decision, out of scope for this PR.

File: tests/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/test_formula_8_85.py (hunk 3 above)
Comment (egarciamendez) [reply tests/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/test_formula_8_85.py:42]: `bool(formula) is expected` matches the existing pattern in 8.58, 8.60, 8.69, 8.70.

File: blueprints/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_85.py (hunk 2 above)
Comment (egarciamendez) [reply blueprints/codes/eurocode/fpr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_85.py:140]: This is the base ComparisonFormula design, float-based, same as 8.41, 8.58, 8.60, 8.69, 8.70. Out of scope for this PR.
